src/campaigns/ai/campaignsAi.service.ts

```diff
@@ -0,0 +1,534 @@
+            calls: 0,
+            yardSigns: 0,
+            events: 0,
+            text: 0,
+            directMail: 0,
+          },
+          thisWeek: {
+            total: 0,
+            doorKnocking: 0,
+            digitalAds: 0,
+            calls: 0,
+            yardSigns: 0,
+            events: 0,
+            text: 0,
+            directMail: 0,
+          },
+          lastWeek: {
+            total: 0,
+            doorKnocking: 0,
+            digitalAds: 0,
+            calls: 0,
+            yardSigns: 0,
+            events: 0,
+            text: 0,
+            directMail: 0,
+          },
+        }
+
+        if (updateHistoryObjects) {
+          for (const update of updateHistoryObjects) {
+            updateHistory.allTime[update.type] += update.quantity
+            updateHistory.allTime.total += update.quantity
+            if (update.createdAt > thisWeek) {
+              updateHistory.thisWeek[update.type] += update.quantity
+              updateHistory.thisWeek.total += update.quantity
+            }
+            if (update.createdAt > twoWeeksAgo && update.createdAt < thisWeek) {
+              updateHistory.lastWeek[update.type] += update.quantity
+              updateHistory.lastWeek.total += update.quantity
+            }
+          }
+        }
+        replaceArr.push({
+          find: 'updateHistory',
+          replace: JSON.stringify(updateHistory),
+        })
+      }
+
+      if (campaign.aiContent) {
+        const {
+          aboutMe,
+          communicationStrategy,
+          messageBox,
+          mobilizing,
+          policyPlatform,
+          slogan,
+          why,
+        } = campaign.aiContent as CampaignAiContent
+        replaceArr.push(
+          {
+            find: 'slogan',
+            replace: slogan?.content,
+          },
+          {
+            find: 'why',
+            replace: why?.content,
+          },
+          {
+            find: 'about',
+            replace: aboutMe?.content,
+          },
+          {
+            find: 'myPolicies',
+            replace: policyPlatform?.content,
+          },
+          {
+            find: 'commStart',
+            replace: communicationStrategy?.content,
+          },
+          {
+            find: 'mobilizing',
+            replace: mobilizing?.content,
+          },
+          {
+            find: 'positioning',
+            replace: messageBox?.content,
+          },
+        )
+      }
+
+      replaceArr.forEach((item) => {
+        try {
+          newPrompt = replaceAll(
+            newPrompt,
+            item.find,
+            item.replace ? item.replace.toString().trim() : '',
+          )
+        } catch (e) {
+          console.log('error at prompt replace', e)
+        }
+      })
+
+      newPrompt += `\n
+        
+      `
+
+      return newPrompt
+    } catch (e) {
+      console.log('Error in helpers/ai/promptReplace', e)
+      //TODO: surface relevant error here
+      return ''
+    }
+  }
+}
```

Comment:
this method _really_ needs to be broken down into the relevant logic pieces regarding how getting a new prompt, produces side-effects in, and depends on, `pathToVictory, campaignPositions, campaignUpdateHistory` and the `user` associated w/ the given campaign.

Seems like, w/ a little thought this could be greatly simplified and would allow each component of the logic flow here to be separated responsibly.